src/components/select-field/BaseSelectField.js

```diff
@@ -224,7 +224,8 @@ class BaseSelectField extends React.Component<Props, State> {
             isOpen &&
             event &&
             event.relatedTarget &&
-            !(event.relatedTarget: window.HTMLInputElement).classList.contains('search-input')
+            !(event.relatedTarget: window.HTMLInputElement).classList.contains('search-input') &&
+            !(event.relatedTarget: window.HTMLInputElement).classList.contains('select-button')
```

Comment:
The reason for this change is as follows:
Since we now have a search input component inside of the selector dropdown, that search input element becomes focused immediately when the dropdown is open. When the user clicks on the Select Field button, this will fire a `handleBlur` call. After the `handleBlur` call is fired, `handleButtonClick` is then fired. The issue that we run into then is, we call `closeDropdown` in `handleBlur`, then `openDropdown` in `handleButtonClick` since in state `isOpen` is already set to `false`. Instead, we only want closeDropdown to be called once, and that should be done in handleButtonClick rather than handleBlur (in the scenario where the user is clicking on the .select-button.

Reply:
This is why the dropdown seems to be stuck open; we close the dropdown then immediately open the dropdown because `handleButtonClick` is fired immediately after `handleBlur`.